From b1d38f603724c585f873ea8551e38eacd2362ba8 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sat, 14 May 2022 22:46:06 -0400
Subject: Return new wxBitmap for GetDisabledBitmap()

This issue was detected by the clang64 compiler; but, it likely needs
fixed for all compilers.

---
 etg/menuitem.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/etg/menuitem.py b/etg/menuitem.py
index ac6cdb38..ba794039 100644
--- a/etg/menuitem.py
+++ b/etg/menuitem.py
@@ -136,11 +136,12 @@ def run():
 
 
     c.find('GetDisabledBitmap').type = 'const wxBitmap*'
+    c.find('GetDisabledBitmap').transferBack = True  # Python takes ownership of the return value
     c.find('GetDisabledBitmap').setCppCode("""\
         #ifdef __WXMSW__
-            return &self->GetDisabledBitmap();
+            return new wxBitmap(self->GetDisabledBitmap());
         #else
-            return &wxNullBitmap;
+            return new wxBitmap(wxNullBitmap);
         #endif
         """)
 
-- 
